src/v/cluster/partition_leaders_table.h

```diff
@@ -193,27 +183,46 @@ class partition_leaders_table {
         model::revision_id partition_revision;
     };
 
-    std::optional<leader_meta>
+    using partition_leaders
+      = absl::node_hash_map<model::partition_id, leader_meta>;
```

Comment:
Sorry in our perf meeting I realized today that we can't really use node_hash_map as even that might do an oversized alloc if we have something like 50k partitions per topic.

However I am now wondering given that this table should contain entries for all the partitions in a topic does anything speak against just using a `chunked_vector` and index by partition id? 

That would avoid the large alloc issue and be even faster?

Reply:
in this particular case the updates are asynchronous and sparse. i.e. getting update for p_id == 10 may happen at any given time when there is no value for p_id == 5. I can think of changing that based on notifications from topics table, but for now i will change this to btree and do it in follow up PR.